lib/schema.js

```diff
@@ -20,6 +20,37 @@ class Schema extends SchemaMetadata {
     return null;
   }
 
+  static concat(...sources) {
```

Comment:
try to use `Schema.protoype.toJSON` to reduce amount of code. You can create new Schema with just typings from `toJSON`